### PR TITLE
Add responds_to hash to OrderEvent

### PR DESCRIPTION
**Pull request by sepans**
To be used in Buyer Counteroffer email sent to seller:

screen shot 2018-12-11 at 4 40 06 pm

### CODE REVIEW

File: app/events/order_event.rb [hunk 1]
```diff
@@ -56,14 +56,21 @@ def last_offer
...
+    end
     {
       id: last_offer.id,
       amount_cents: last_offer.amount_cents,
       shipping_total_cents: last_offer.shipping_total_cents,
       tax_total_cents: last_offer.tax_total_cents,
       from_participant: last_offer.from_participant,
       creator_id: last_offer.creator_id,
-      responds_to: last_offer.responds_to_id
+      responds_to: responds_to
```
Comment (joeyAghion) [inline app/events/order_event.rb]: This is very similar to the `respond_to?` method built-in to every ruby object. 😬

That maybe isn't enough to demand a change, but it does make me wonder if there's a better name, like `in_response_to`.

File: app/events/order_event.rb [hunk 2]
```diff
@@ -56,14 +56,21 @@ def last_offer
     return unless @object.last_offer
 
     last_offer = @object.last_offer
+    in_response_to = nil
```
Comment (ashkan18) [inline app/events/order_event.rb]: #minor you can change this to:

```ruby
in_response_to = if last_offer.responds_to
{ 
    id: last_offer.responds_to.id,
    amount_cents: last_offer.responds_to.amount_cents,
    created_at: last_offer.responds_to.created_at,
    from_participant: last_offer.responds_to.from_participant
}
end
```

File: spec/events/order_event_spec.rb [hunk 3]
```diff
@@ -124,7 +124,20 @@
         expect(event.properties[:last_offer][:tax_total_cents]).to eq 40_00
         expect(event.properties[:last_offer][:from_participant]).to eq 'seller'
         expect(event.properties[:last_offer][:creator_id]).to eq 'partner-admin'
-        expect(event.properties[:last_offer][:responds_to_id]).to be_nil
+        expect(event.properties[:last_offer][:responds_to]).to be_nil
```
Comment (yuki24) [inline spec/events/order_event_spec.rb]: @sepans Is this supposed to be `in_response_to `?

cc @joeyAghion

File: spec/events/order_event_spec.rb [hunk 4]
```diff
@@ -124,7 +124,20 @@
...
+      end
+    end
+    context 'with last_offer that responds to another offer' do
+      let(:previous_offer) { Fabricate(:offer, order: order, amount_cents: 100_00, shipping_total_cents: 50_00, tax_total_cents: 20_00, from_id: user_id, from_type: 'user', creator_id: user_id) }
+      let(:offer) { Fabricate(:offer, order: order, amount_cents: 200_00, shipping_total_cents: 100_00, tax_total_cents: 40_00, from_id: seller_id, from_type: 'gallery', creator_id: 'partner-admin', responds_to: previous_offer) }
```
Comment (yuki24) [inline spec/events/order_event_spec.rb]: Same here

**Pull request merged**